test/blpapi.test.ts

```diff
@@ -0,0 +1,616 @@
+            session.subscribe([ subscription ]).catch((err: Error): void => {
+                should(err).not.be.empty;
+                err.name.should.eql('BlpApiError');
+                err.message.should.eql('//blp/mktbar Service Fail to Open.');
+                done();
+            });
+        });
+    });
+
+    describe('#unsubscribe()', (): void => {
+        it('should unsubscribed all successfully', (done: Function): void => {
+            instructions.onOpenService = (mockSession: MOCKBLPAPIJS.Session,
+                                          uri: string,
+                                          cid: number): void => {
+                mockSession.sendServiceOpened(cid);
+            };
+            var mSession: MOCKBLPAPIJS.Session;
+            var sub: { event: string; cid: number };
+            instructions.onSubscribe = (mockSession: MOCKBLPAPIJS.Session,
+                                        subs: { event: string; cid: number }[]): void => {
+                mSession = mockSession;
+                sub = subs[0];
+                mockSession.sendSubscriptionData(sub.event, sub.cid);
+            };
+            var subscription = new BLPAPI.Subscription('ABC', ['P']);
+            session.subscribe([subscription]);
+            subscription.once('data', (data: any): void => {
+                data.should.eql('TestData');
+                session.unsubscribe([subscription]);
+                mSession.sendSubscriptionData(sub.event, sub.cid);
+                subscription.once('data', (data: any): void => {
+                    done(new Error('unsubscribe fails'));
+                });
+                setTimeout(done, 100);
+            });
+        });
+        it('should unsubscribed part successfully', (done: Function): void => {
+            instructions.onOpenService = (mockSession: MOCKBLPAPIJS.Session,
+                                          uri: string,
+                                          cid: number): void => {
+                mockSession.sendServiceOpened(cid);
+            };
+            var mSession: MOCKBLPAPIJS.Session;
+            var sub1: { event: string; cid: number };
+            var sub2: { event: string; cid: number };
+            instructions.onSubscribe = (mockSession: MOCKBLPAPIJS.Session,
+                                        subs: { event: string; cid: number }[]): void => {
+                mSession = mockSession;
+                sub1 = subs[0];
+                sub2 = subs[1];
+                mockSession.sendSubscriptionData(sub1.event, sub1.cid);
+                mockSession.sendSubscriptionData(sub2.event, sub2.cid);
+            };
+            var subscription1 = new BLPAPI.Subscription('ABC', ['P']);
+            var subscription2 = new BLPAPI.Subscription('DEF', ['P']);
+            var counter: number = 0;
+            session.subscribe([ subscription1, subscription2 ]);
+            subscription1.on('data', (data: any): void => {
```

Comment:
You don't really check that `unsubscribe` worked on `subscription2` since you don't have a listener on it.